### PR TITLE
feat(igxPivotGrid): Autogenerate pivot config option.

### LINKED ISSUE #12366: Autogenerate dimensions and values from the data when no pivot configuration is defined
## Is your feature request related to a problem? Please describe.
Pivot configuration can be difficult to be defined. Autogenerating simple non-hierarchical dimensions and values from  fields of the data when no pivot configuration is defined would be a great benefit for the users. 

## Describe the solution you'd like
Additional field or a place in the data selector where those fields can be populated along side with an empty pivot grid and allow the users to pivot with the data using those autogenerated dimensions/values.

### PR DESCRIPTION
**Pull request by MayaKirova**
Closes #12366  
Spec is here: https://github.com/IgniteUI/igniteui-angular/wiki/igxPivotGrid-Specification#autogenerate-configuration

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by skrustev — APPROVED** [review]

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts [hunk 1]
```diff
@@ -2206,7 +2207,7 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
         const columnDimensions: IPivotDimension[] = [];
         const rowDimensions: IPivotDimension[] = [];
         const values: IPivotValue[] = [];
-
+        let isFirstDate = false;
```
Comment (skrustev) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts:2210]: Shouldn't this be true?

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts [hunk 2]
```diff
@@ -2228,12 +2229,15 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
                 }
             case "date":
```
Comment (skrustev) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts:2230]: I was wondering if maybe would be nice to also include strings that represent dates, but that might be not a good idea to assume.

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts [hunk 3]
```diff
@@ -2228,12 +2229,15 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
...
             {
-                const dimension: IPivotDimension = {
```
Comment (skrustev) [inline projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts]: There's a bug in the `IgxPivotDateDimension` that doesn't pass the enabled property even to the root dimension it creates.

**Review by skrustev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts (hunk 2 above)
Comment (MayaKirova) [reply projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts:2230]: I was wondering about that too, but the auto-gen of columns also does not do that. It only generates date columns if the field has a Date object. So I left it like that for the pivot config as well. 
Technically there's an init event for the dimensions so if someone wants to mark a specific dimension a different type or enable/disable it before rendering, they could.

**Review by skrustev — APPROVED** [review]

**Review by skrustev — APPROVED** [review]

**Pull request merged**